package/MDAnalysis/lib/include/iterators.h

```diff
@@ -0,0 +1,142 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode:nil; -*- */
+/* vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4 */
+/*
+  MDAnalysis --- http://mdanalysis.googlecode.com
+
+  Copyright (c) 2006-2014 Naveen Michaud-Agrawal,
+                Elizabeth J. Denning, Oliver Beckstein,
+                and contributors (see AUTHORS for the full list)
+  Released under the GNU Public Licence, v2 or any higher version
+
+  Please cite your use of MDAnalysis in published work:
+
+      N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and
+      O. Beckstein. MDAnalysis: A Toolkit for the Analysis of
+      Molecular Dynamics Simulations. J. Comput. Chem. 32 (2011), 2319--2327,
+      in press.
+*/
+
+#ifndef __ITERATORS_H
+#define __ITERATORS_H
+
+#include <cstdint>
+#include <vector>
+#include <cstdio>
+
+/* This header provides classes to iterate groups (eg AtomGroup) and NumPy
+   arrays using the same API. This is done by using thin wrapper classes with
+   the same interface methods:
+
+   - reset_iteration() -> rewind to start of group/array
+   - load_into_external_buffer -> pass values from group/array into an external
+     buffer (likely a stack allocated array)
+   - seek (optional/unused) -> go to a particular atom in the group/array
```

Comment:
yagni?

Reply:
Might need to fix overhanging part of array but will investigate.

Reply:
Needed in the end.